### DISCUSSION TITLE
does supabase support transactions like firebase?

### DISCUSSION THREAD

**Discussion by SalishKumar** [supabase/supabase#14796] [Questions]
Hi,
I have a json file, i will be parsing  that and writing it to the supabase in forms of 3 tables, so i want data to be consistent in 3 tables, but i am not able to find something relatable in supabase documentation

**Comment by GaryAustin1** [comment]
You would do all the operations in a Postgres function and call with rpc.   Functions are run as a transaction with rollback on any error.

**Reply by SalishKumar** [reply]
thanks, but do we have similar interface like firebase?

**Reply by GaryAustin1** [reply]
Sorry I was not clear.  No.  There is no ability to do multiple API calls as a transaction.  The rpc call to do a transaction in a function with multiple SQL calls is the way.

**Comment by vickkhera** [comment]
If you use a native Postgres connection, you can use the transactions like normal there. It's just not possible via the API.
